lib/xcode/install.rb

```diff
@@ -58,16 +59,31 @@ def current_symlink
     end
 
     def download(version, progress, url = nil)
-      return unless url || exist?(version)
-      xcode = seedlist.find { |x| x.name == version } unless url
+      xcode = find_xcode_version(version) if url.nil?
+      return if url.nil? && xcode.nil?
```

Comment:
this is a good opportunity to do this piece once: `url || xcode.path` and then use it a few times in this method.
Maybe call it `fetch_url` or something instead of relying on `url || xcode.path` in a few places.

Reply:
Doesn't look like that's possible, as we're accessing `xcode.path` (file path) and `xcode.url` (download URL), so those are not the same thing. While this could be optimized by adding an extra `if else` block, it doesn't seem to really improve the quality of code in this scenario